File: Data Structures & Algorithms/max-water-container/submission-0.cs
public class Solution {
    public int MaxArea(int[] nums)
    {
        var longestArea = 0;
        var length = nums.Length;
        for (var i = 0; i < length; i++)
        {
            for (var j = i + 1; j < length; j++)
            {
                var area = Math.Min(nums[i], nums[j]) * Math.Abs(j - i);
                longestArea = Math.Max(area, longestArea);
            }
        }

        return longestArea;
    }

}

File: Data Structures & Algorithms/trapping-rain-water/submission-0.go
import "slices"

func trap(height []int) int {
	totalWater := 0
	for i, _ := range height {
		area := colArea(i, height)
		totalWater += area
	}
	return totalWater
}

func colArea(i int, nums []int) int {
	if i == 0 || i == len(nums)-1 {
		return 0
	}

	leftLargest := slices.Max(nums[:i])
	rightLargest := slices.Max(nums[i+1:])

	return max(min(leftLargest, rightLargest)-nums[i], 0)
}
